teamcity: failed test: TestStoreReplicaVisitor

The following tests appear to have failed on master (testrace): TestStoreReplicaVisitor

You may want to check [for open issues](https://github.com/cockroachdb/cockroach/issues?q=is%3Aissue+is%3Aopen+TestStoreReplicaVisitor).

[#976182](https://teamcity.cockroachdb.com/viewLog.html?buildId=976182):


```
TestStoreReplicaVisitor
... 0x3cc2cc0, 0xc4210455c0, 0xc420016240, 0x29, 0x0, 0x0, 0xc4203f2840)
	/go/src/github.com/cockroachdb/cockroach/pkg/util/stop/stopper.go:324 +0xf4
created by github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).RunAsyncTask
	/go/src/github.com/cockroachdb/cockroach/pkg/util/stop/stopper.go:319 +0x14d

goroutine 34154 [semacquire]:
sync.runtime_notifyListWait(0xc4216f94d0, 0x23)
	/usr/local/go/src/runtime/sema.go:510 +0x11a
sync.(*Cond).Wait(0xc4216f94c0)
	/usr/local/go/src/sync/cond.go:56 +0x8e
github.com/cockroachdb/cockroach/pkg/storage.(*raftScheduler).worker(0xc420a9e900, 0x3cc2cc0, 0xc420c8d5c0)
	/go/src/github.com/cockroachdb/cockroach/pkg/storage/scheduler.go:196 +0x9c
github.com/cockroachdb/cockroach/pkg/storage.(*raftScheduler).Start.func2(0x3cc2cc0, 0xc420c8d5c0)
	/go/src/github.com/cockroachdb/cockroach/pkg/storage/scheduler.go:165 +0x56
github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).RunWorker.func1(0xc420570700, 0xc420a68510, 0xc4205706f0)
	/go/src/github.com/cockroachdb/cockroach/pkg/util/stop/stopper.go:199 +0x14b
created by github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).RunWorker
	/go/src/github.com/cockroachdb/cockroach/pkg/util/stop/stopper.go:192 +0xbb

goroutine 34153 [semacquire]:
sync.runtime_notifyListWait(0xc4216f94d0, 0x22)
	/usr/local/go/src/runtime/sema.go:510 +0x11a
sync.(*Cond).Wait(0xc4216f94c0)
	/usr/local/go/src/sync/cond.go:56 +0x8e
github.com/cockroachdb/cockroach/pkg/storage.(*raftScheduler).worker(0xc420a9e900, 0x3cc2cc0, 0xc4204942a0)
	/go/src/github.com/cockroachdb/cockroach/pkg/storage/scheduler.go:196 +0x9c
github.com/cockroachdb/cockroach/pkg/storage.(*raftScheduler).Start.func2(0x3cc2cc0, 0xc4204942a0)
	/go/src/github.com/cockroachdb/cockroach/pkg/storage/scheduler.go:165 +0x56
github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).RunWorker.func1(0xc4205706d0, 0xc420a68510, 0xc4205706c0)
	/go/src/github.com/cockroachdb/cockroach/pkg/util/stop/stopper.go:199 +0x14b
created by github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).RunWorker
	/go/src/github.com/cockroachdb/cockroach/pkg/util/stop/stopper.go:192 +0xbb

goroutine 34112 [semacquire]:
sync.runtime_notifyListWait(0xc4216f94d0, 0x9)
	/usr/local/go/src/runtime/sema.go:510 +0x11a
sync.(*Cond).Wait(0xc4216f94c0)
	/usr/local/go/src/sync/cond.go:56 +0x8e
github.com/cockroachdb/cockroach/pkg/storage.(*raftScheduler).worker(0xc420a9e900, 0x3cc2cc0, 0xc420c8cab0)
	/go/src/github.com/cockroachdb/cockroach/pkg/storage/scheduler.go:196 +0x9c
github.com/cockroachdb/cockroach/pkg/storage.(*raftScheduler).Start.func2(0x3cc2cc0, 0xc420c8cab0)
	/go/src/github.com/cockroachdb/cockroach/pkg/storage/scheduler.go:165 +0x56
github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).RunWorker.func1(0xc4205702e0, 0xc420a68510, 0xc4205702d0)
	/go/src/github.com/cockroachdb/cockroach/pkg/util/stop/stopper.go:199 +0x14b
created by github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).RunWorker
	/go/src/github.com/cockroachdb/cockroach/pkg/util/stop/stopper.go:192 +0xbb

goroutine 34103 [semacquire]:
sync.runtime_notifyListWait(0xc4216f94d0, 0x1)
	/usr/local/go/src/runtime/sema.go:510 +0x11a
sync.(*Cond).Wait(0xc4216f94c0)
	/usr/local/go/src/sync/cond.go:56 +0x8e
github.com/cockroachdb/cockroach/pkg/storage.(*raftScheduler).worker(0xc420a9e900, 0x3cc2cc0, 0xc420c8c840)
	/go/src/github.com/cockroachdb/cockroach/pkg/storage/scheduler.go:196 +0x9c
github.com/cockroachdb/cockroach/pkg/storage.(*raftScheduler).Start.func2(0x3cc2cc0, 0xc420c8c840)
	/go/src/github.com/cockroachdb/cockroach/pkg/storage/scheduler.go:165 +0x56
github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).RunWorker.func1(0xc420570150, 0xc420a68510, 0xc420570140)
	/go/src/github.com/cockroachdb/cockroach/pkg/util/stop/stopper.go:199 +0x14b
created by github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).RunWorker
	/go/src/github.com/cockroachdb/cockroach/pkg/util/stop/stopper.go:192 +0xbb




```

Please assign, take a look and update the issue accordingly.

```
panic: runtime error: invalid memory address or nil pointer dereference [recovered]
	panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x15b271f]

goroutine 34119 [running]:
github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).Recover(0xc420a68510, 0x3cc2cc0, 0xc421045980)
	/go/src/github.com/cockroachdb/cockroach/pkg/util/stop/stopper.go:184 +0x12e
panic(0x3148ae0, 0x4d45b00)
	/usr/local/go/src/runtime/panic.go:502 +0x24a
github.com/cockroachdb/cockroach/vendor/go.etcd.io/etcd/raft.(*Progress).maybeUpdate(...)
	/go/src/github.com/cockroachdb/cockroach/vendor/go.etcd.io/etcd/raft/progress.go:120
github.com/cockroachdb/cockroach/vendor/go.etcd.io/etcd/raft.(*raft).appendEntry(0xc420133cc0, 0xc42017c550, 0x1, 0x1)
	/go/src/github.com/cockroachdb/cockroach/vendor/go.etcd.io/etcd/raft/raft.go:646 +0x20f
github.com/cockroachdb/cockroach/vendor/go.etcd.io/etcd/raft.(*raft).becomeLeader(0xc420133cc0)
	/go/src/github.com/cockroachdb/cockroach/vendor/go.etcd.io/etcd/raft/raft.go:742 +0x2ac
github.com/cockroachdb/cockroach/vendor/go.etcd.io/etcd/raft.(*raft).campaign(0xc420133cc0, 0x3443ea9, 0x10)
	/go/src/github.com/cockroachdb/cockroach/vendor/go.etcd.io/etcd/raft/raft.go:765 +0x8be
github.com/cockroachdb/cockroach/vendor/go.etcd.io/etcd/raft.(*raft).campaign(0xc420133cc0, 0x3449bc0, 0x13)
	/go/src/github.com/cockroachdb/cockroach/vendor/go.etcd.io/etcd/raft/raft.go:763 +0x924
github.com/cockroachdb/cockroach/vendor/go.etcd.io/etcd/raft.(*raft).Step(0xc420133cc0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, ...)
	/go/src/github.com/cockroachdb/cockroach/vendor/go.etcd.io/etcd/raft/raft.go:890 +0x1aa2
github.com/cockroachdb/cockroach/vendor/go.etcd.io/etcd/raft.(*RawNode).Campaign(0xc4216f99c0, 0xc4216005d0, 0x3)
	/go/src/github.com/cockroachdb/cockroach/vendor/go.etcd.io/etcd/raft/rawnode.go:137 +0x87
github.com/cockroachdb/cockroach/pkg/storage.(*Replica).maybeCampaignOnWakeLocked(0xc4203c9c00, 0x3cc2cc0, 0xc4216005d0)
	/go/src/github.com/cockroachdb/cockroach/pkg/storage/replica.go:627 +0x5a6
github.com/cockroachdb/cockroach/pkg/storage.(*Replica).withRaftGroupLocked(0xc4203c9c00, 0x3579401, 0x3577698, 0xc420bda8a0, 0x10f96eb)
	/go/src/github.com/cockroachdb/cockroach/pkg/storage/replica.go:556 +0x57b
github.com/cockroachdb/cockroach/pkg/storage.(*Replica).maybeInitializeRaftGroup(0xc4203c9c00, 0x3cc2cc0, 0xc421045980)
	/go/src/github.com/cockroachdb/cockroach/pkg/storage/replica.go:1952 +0x157
github.com/cockroachdb/cockroach/pkg/storage.(*Replica).sendWithRangeID(0xc4203c9c00, 0x3cc2cc0, 0xc421045980, 0x6, 0x7b, 0x27, 0x0, 0x0, 0x6, 0x0, ...)
	/go/src/github.com/cockroachdb/cockroach/pkg/storage/replica.go:1993 +0x195
github.com/cockroachdb/cockroach/pkg/storage.(*Replica).Send(0xc4203c9c00, 0x3cc2cc0, 0xc421045980, 0x7b, 0x27, 0x0, 0x0, 0x6, 0x0, 0x0, ...)
	/go/src/github.com/cockroachdb/cockroach/pkg/storage/replica.go:1965 +0xb6
github.com/cockroachdb/cockroach/pkg/storage.(*pendingLeaseRequest).requestLeaseAsync.func2(0x3cc2cc0, 0xc421045980)
	/go/src/github.com/cockroachdb/cockroach/pkg/storage/replica_range_lease.go:323 +0x6c1
github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).RunAsyncTask.func1(0xc420a68510, 0x3cc2cc0, 0xc421045980, 0xc420054240, 0x35, 0x3cf7980, 0xc4202b36c0, 0xc420a9fe60)
	/go/src/github.com/cockroachdb/cockroach/pkg/util/stop/stopper.go:324 +0xf4
created by github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).RunAsyncTask
	/go/src/github.com/cockroachdb/cockroach/pkg/util/stop/stopper.go:319 +0x14d
```

A nil pointer dereference on a `*raft.Progress`. That's surprising.

--- a/pkg/storage/store_test.go
+++ b/pkg/storage/store_test.go
@@ -176,9 +176,10 @@ func createTestStoreWithoutStart(t testing.TB, stopper *stop.Stopper, cfg *Store
 	// so just disable the scanner for all tests that use this function
 	// instead of figuring out exactly which tests need it.
 	cfg.TestingKnobs.DisableScanner = true
-	// The scanner affects background operations; we must also disable
-	// the split queue separately to cover event-driven splits.
+	// The scanner affects background operations; we must also disable the split
+	// and merge queues separately to cover event-driven splits and merges.
 	cfg.TestingKnobs.DisableSplitQueue = true
+	cfg.TestingKnobs.DisableMergeQueue = true
 	eng := engine.NewInMem(roachpb.Attributes{}, 10<<20)
 	stopper.AddCloser(eng)
 	cfg.Transport = NewDummyRaftTransport(cfg.Settings)

